tests/testsuite/artifact_dep.rs

```diff
@@ -777,19 +784,22 @@ fn build_script_with_selected_dashed_bin_artifact_and_lib_true() {
     let build_script_output = build_script_output_string(&p, "foo");
     // we need the binary directory for this artifact and the binary itself
     if cfg!(target_env = "msvc") {
-        compare::assert_match_exact(
-            &format!(
-                "[..]/artifact/bar-baz-[..]/bin\n\
-                 [..]/artifact/bar-baz-[..]/bin/baz_suffix{}",
-                std::env::consts::EXE_SUFFIX,
-            ),
+        assert_e2e().eq(
             &build_script_output,
+            str![[r#"
+[ROOT]/foo/target/debug/deps/artifact/bar-baz-[..]/bin
+[ROOT]/foo/target/debug/deps/artifact/bar-baz-[..]/bin/baz_suffix[EXE]
+
+"#]],
         );
     } else {
-        compare::assert_match_exact(
-            "[..]/artifact/bar-baz-[..]/bin\n\
-        [..]/artifact/bar-baz-[..]/bin/baz_suffix-[..]",
+        assert_e2e().eq(
             &build_script_output,
+            str![[r#"
+[ROOT]/foo/target/debug/deps/artifact/bar-baz-[..]/bin
+[ROOT]/foo/target/debug/deps/artifact/bar-baz-[..]/bin/baz_suffix-[..]
+
```

Comment:
Can I ask when we need to add a blank line? I tried this locally, it seemed to pass the test after I removed it.
<img width="997" alt="image" src="https://github.com/rust-lang/cargo/assets/29879298/b3cdcf96-d903-4e99-ae32-7f33d8ff8df1">

Reply:
Got it.